Add hreflang attributes to make Translations easier

bazzadp opened the pull request:
We encourage adding `hreflang="en"` to links for translators as discussed here: https://github.com/orgs/HTTPArchive/teams/translators/discussions/4

In #1793 this added a `(en)` superscript icon to tell the reader the link is to an English language page, which is an [web quality best practice](https://checklists.opquast.com/en/web-quality-assurance/the-main-language-of-a-links-target-page-is-provided-if-different-from-the-original-page):

![Screenshot of French Accessibility chapter with language icons on links](https://user-images.githubusercontent.com/10931297/109493744-d6073380-7a84-11eb-9072-0e3e65963019.png)

However this is an extra burden for the translator to add this attribute, and also in many places convert markdown links to HTML links to allow them to add this attribute. It's also not needed for some links (e.g. Wikipedia and MDN) as we automatically add the icon based on the URL. But this is an extra thing for translators to be aware off.

So I applied some regex's to update the English versions (and those translations that have already happened) to make this easier:

**Convert all markdown links:**
Search: `\[([^\]]*)\]\((https:\/\/[^\)]*)\)`
Replace: `<a hreflang="en" href="$2">$1</a>`
Files to include: `content:`

**Add hreflang to all HTML links:**
Search: `<a href="`
Replace: `<a hreflang="en" href="`
Files to include: `content:`

**Remove any double hreflang attributes:**
Search: `hreflang="[a-z]*" ([^>]*) hreflang="([a-z]*)"([^>]*)>`
Replace: `hreflang="$2" $1 $3>`
Files to include: `content:`

**Revert back the links that don't need this:**
Search: `<a hreflang="en" href="https://twitter.com`
Replace: `<a href="https://twitter.com`
Files to include: `content:`

Search: `<a href="(https://[a-z]*.wikipedia.org[^"]*)">([^<]*)</a>`
Replace: `[$2]($1)`
Files to include: `content:`

Search: `<a hreflang="en" href="(https://developer.mozilla.org[^"]*">([^<]*)</a>`
Replace: `[$2]($1)`
Files to include: `content:`

noah-vdv reviewed (approved):
Awesome!

noah-vdv commented:
Just one question: I see you've now also changed links like `nl.wikipedia` to html. Should we do that from now on for all translated links?

bazzadp commented:
> Just one question: I see you've now also changed links like `nl.wikipedia` to html. Should we do that from now on for all translated links?

Nope, my mistake (regexes are tricky!). Fixed now.

noah-vdv commented on src/content/nl/2020/jamstack.md:
```diff
@@ -7,7 +7,7 @@ reviewers: [MaedahBatool, phacks]
 editors: [bazzadp]
 translators: [noah-vdv]
-ahmadawais_bio: Ahmad Awais is een bekroonde open-source ingenieur, Google Developers Expert Dev Voorstander, Node.js Community Committee Outreach Leider, WordPress Core Dev en VP Engineering DevRel bij WGA. Hij heeft verschillende open-source softwaretools geschreven die door miljoenen ontwikkelaars over de hele wereld worden gebruikt. Zoals zijn <a hreflang="en" href="https://shadesofpurple.pro ...
+ahmadawais_bio: Ahmad Awais is een bekroonde open-source ingenieur, Google Developers Expert Dev Voorstander, Node.js Community Committee Outreach Leider, WordPress Core Dev en VP Engineering DevRel bij WGA. Hij heeft verschillende open-source softwaretools geschreven die door miljoenen ontwikkelaars over de hele wereld worden gebruikt. Zoals zijn <a hreflang="en" href="https://shadesofpurple.pro ...
```
nit (remove some whitespace at "cursussen" and "NODE cli" (don't actually know if it's useful but oh well))
```suggestion
ahmadawais_bio: Ahmad Awais is een bekroonde open-source ingenieur, Google Developers Expert Dev Voorstander, Node.js Community Committee Outreach Leider, WordPress Core Dev en VP Engineering DevRel bij WGA. Hij heeft verschillende open-source softwaretools geschreven die door miljoenen ontwikkelaars over de hele wereld worden gebruikt. Zoals zijn <a hreflang="en" href="https://shadesofpurple.pro/more">Shades of Purple</a> codethema of projecten zoals de <a hreflang="en" href="https://github.com/AhmadAwais/corona-cli">corona-cli</a>. Awais houdt van lesgeven. Meer dan 20.000 ontwikkelaars leren van zijn <a hreflang="en" href="https://AhmadAwais.com/courses/">cursussen</a>, dwz <a hreflang="en" href="https://nodecli.com/">Node CLI</a>, <a hreflang="en" href="https://vscode.pro/">VSCode.pro</a>, en <a hreflang="en" href="https://nextjsbeginner.com/">Next.js Beginner</a>. Awais ontving de erkenning van FOSS-gemeenschapsleiderschap als een van de <a hreflang="en" href="https://ahmadawais.com/github-stars/">12 aanbevolen GitHub-sterren</a>. Hij is lid van het Smashing Magazine Experts Panel; aanbevolen &amp; gepubliceerde auteur bij CSS-Tricks, Tuts+, Scotch.io, SitePoint. Je kunt hem meestal vinden op Twitter <a href="https://twitter.com/MrAhmadAwais/">@MrAhmadAwais</a> waar hij zijn <a hreflang="en" href="https://awais.dev/odmt">#OneDevMinute</a> ontwikkelaarstips deelt.
```

tunetheweb reviewed (commented):
(ignore - commented on wrong issue)
